Uplift/Utils/Custom/StretchyModifier.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  StretchyHeaderModifier.swift
+//  Uplift
+//
+//  Created by Caitlyn Jin on 2/11/24.
+//  Copyright © 2024 Cornell AppDev. All rights reserved.
+//
+
+import SwiftUI
+
+/// A view modifier that applies a stretchy effect to the header image.
+struct StretchyModifier: ViewModifier {
+
+    var geometryProxy: GeometryProxy
+
+    func body(content: Content) -> some View {
+        let minY = geometryProxy.frame(in: .global).minY
+        let originY = geometryProxy.frame(in: .global).origin.y
+        let height = geometryProxy.size.height
+
+        content
+            .frame(height: minY > 0 ? height + originY : height)
```

Comment:
what does this logic do? I think that if some vertical delta component is non-negative, we change our height, otherwise we keep it the same. Is the purpose of this for making the header have more "range"/more animated?

Reply:
Yeah, the `content` in this case is the image itself, and we are adjusting the height of the image based on the view's frame relative to the whole screen. If we scroll up, the image is lower on the screen, so we increase the height of the image and the `.offset` modifier sets the image's y-offset so that it stays at the top of the screen. This creates an "animated stretchy" effect.

Reply:
+1